File: GraffiTab_iOS_SDK/Classes/Network/Session/GTCookieManager.m
//
//  CookieManager.m
//  GraffiTab-iOS
//
//  Created by Georgi Christov on 27/11/2014.
//  Copyright (c) 2014 GraffiTab. All rights reserved.
//

#import "GTCookieManager.h"

@implementation GTCookieManager

+ (void)saveCookies {
    NSMutableArray *cookieArray = [[NSMutableArray alloc] init];
    
    for (NSHTTPCookie *cookie in [[NSHTTPCookieStorage sharedHTTPCookieStorage] cookies]) {
        [cookieArray addObject:cookie.name];
        
        NSMutableDictionary *cookieProperties = [NSMutableDictionary dictionary];
        [cookieProperties setObject:cookie.name forKey:NSHTTPCookieName];
        [cookieProperties setObject:cookie.value forKey:NSHTTPCookieValue];
        [cookieProperties setObject:cookie.domain forKey:NSHTTPCookieDomain];
        [cookieProperties setObject:cookie.path forKey:NSHTTPCookiePath];
        [cookieProperties setObject:[NSNumber numberWithLong:cookie.version] forKey:NSHTTPCookieVersion];
        
        [cookieProperties setObject:[[NSDate date] dateByAddingTimeInterval:2629743] forKey:NSHTTPCookieExpires];
        
        [[NSUserDefaults standardUserDefaults] setValue:cookieProperties forKey:cookie.name];
        [[NSUserDefaults standardUserDefaults] synchronize];
    }
    
    [[NSUserDefaults standardUserDefaults] setValue:cookieArray forKey:@"cookieArray"];
    [[NSUserDefaults standardUserDefaults] synchronize];
}

+ (void)loadCookies {
    NSMutableArray* cookieDictionary = [[NSUserDefaults standardUserDefaults] valueForKey:@"cookieArray"];
    
    for (int i=0; i < cookieDictionary.count; i++) {
        NSMutableDictionary* cookieDictionary1 = [[NSUserDefaults standardUserDefaults] valueForKey:[cookieDictionary objectAtIndex:i]];
        NSHTTPCookie *cookie = [NSHTTPCookie cookieWithProperties:cookieDictionary1];
        [[NSHTTPCookieStorage sharedHTTPCookieStorage] setCookie:cookie];
    }
}

@end

File: GraffiTab-iOS-SDK/Managers/GTCookieManager.swift
//
//  GTCookieManager.swift
//  GraffiTab-iOS-SDK
//
//  Created by Georgi Christov on 04/04/2016.
//  Copyright © 2016 GraffiTab. All rights reserved.
//

import UIKit

class GTCookieManager: NSObject {

    class func saveCookies() {
        let cookieStorage = NSHTTPCookieStorage.sharedHTTPCookieStorage()
        let defaults = NSUserDefaults.standardUserDefaults()
        let cookiesArray = NSMutableArray()
        
        for cookie in cookieStorage.cookies! {
            cookiesArray.addObject(cookie.properties!)
        }
        
        defaults.setObject(cookiesArray, forKey: GTCookieConstants.CookiesKey)
    }
    
    class func loadCookies() {
        let cookieStorage = NSHTTPCookieStorage.sharedHTTPCookieStorage()
        let defaults = NSUserDefaults.standardUserDefaults()
        
        let cookiesArray = defaults.arrayForKey(GTCookieConstants.CookiesKey)
        if (cookiesArray != nil) {
            for cookieProperties in cookiesArray! {
                let cookie = NSHTTPCookie(properties: cookieProperties as! [String : AnyObject])
                cookieStorage.setCookie(cookie!)
            }
        }
    }
}
